server/src/main/java/org/elasticsearch/index/query/RangeQueryBuilder.java

```diff
@@ -267,18 +254,22 @@ public RangeQueryBuilder timeZone(String timeZone) {
         if (timeZone == null) {
             throw new IllegalArgumentException("timezone cannot be null");
         }
-        this.timeZone = DateTimeZone.forID(timeZone);
+        try {
+            this.timeZone = ZoneId.of(timeZone);
+        } catch (ZoneRulesException e) {
```

Comment:
`ZoneId#of()` can also throw `DateTimeException` if the zone id is invalid. Can we guarantee at this point that this is impossible or why are we only catching `ZoneRulesException`?

Reply:
catching DateTimeException makes more sense here